### PR TITLE
Implement 'POST /connect-all?wait=...&iface=...'

### PR DESCRIPTION
**Pull request by ptabor**
- **Implement 'POST /connect-all?wait=...&iface=...'**
- **Added autoconnect option.**

### CODE REVIEW

**Comment by sfc-gh-ptabor** [comment]
Cannot edit the description, but wanted to say there: 

`It's a common use-case to connect to all available chromecasts when you have local server.`

**Review by vishen — COMMENTED** [review]
This looks good to me, just one clarification then I will merge.

File: http/handlers.go [hunk 1]
```diff
@@ -43,10 +45,18 @@ func (h *Handler) Serve(addr string) error {
 	return http.ListenAndServe(addr, h)
 }
 
+// Autoconnect configures the handler to perform auto-discovery of all the cast devices & groups.
+func (h *Handler) Autoconnect() error {
```
[the remark this answers is not in the document]
Comment (sfc-gh-ptabor) [reply http/handlers.go]: It's to be registered side by side to NewHandler. For http -> there is '/connect-all`.

Added comment and moved close to the NewHandler. 
Considered changing `NewHandler` signature, but that would be a breaking change.

File: http/handlers.go [hunk 2]
```diff
@@ -201,28 +213,80 @@ func (h *Handler) connect(w http.ResponseWriter, r *http.Request) {
...
+		return nil, err
 	}
+	return app, nil
+}
+
+func (h *Handler) connectAll(w http.ResponseWriter, r *http.Request) {
+	h.log("connecting all devices")
+	q := r.URL.Query()
+
+	wait := q.Get("wait")
```
Comment (vishen) [inline http/handlers.go:251]: And here

File: http/handlers.go [hunk 3]
```diff
@@ -45,8 +55,9 @@ func (h *Handler) Serve(addr string) error {
 
 func (h *Handler) registerHandlers() {
 	/*
-		GET /devices
+		GET /devices?wait=...&iface=...
```
Comment (vishen) [inline http/handlers.go:58]: Should all the `wait`'s actually be "port"?

File: http/handlers.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (vishen) [reply http/handlers.go]: Gotcha, makes sense. I think this way is fine :)

**Comment by vishen** [comment]
Sorry ignore my comments, misread the code.

**Pull request merged**